Updates compileOF.sh for Linux to solve `chown` error

xuv opened the pull request:
Very small fix. Solving errors when compiling using a different user than the one logged in.

Example, compilation would produce error in this case:
 1. log into a machine using a certain user `user1`
 2. use sudo powers from user1 to switch to `user2`
 3. launch `./compileOF.sh` as user2
 4. breaks when reaching the line `chown -R $ID:$GROUP_ID ../lib/*` because the `$ID` returned by `logname` earlier in the script is `user1` and not `user2` who is calling the script. 

Using `whoami` instead of `logname` fixes this error without breaking the script for other common cases.

arturoc commented:
this was done like this cause the install_dependencies script which is run under sudo was running compile_OF at the end so it needed the original user to change the owner. now install_dependencies doesn't run compile_OF anymore so it's safe to just remove this lines completely from compile_OF.sh.

can you change the PR to that?

thanks!

xuv commented:
@arturoc Thx for the explanation. I have modified the code to reflect this. I went further than just removing that one last line and also removed all the calls to the same type of action. I'll be happy to revert that change if you think this is wrong.

arturoc commented:
thanks!
